tests/python/contrib/test_hexagon/benchmark_util.py

```diff
@@ -0,0 +1,148 @@
+    """
+    BUILTIN_COLUMN_NAMES = set([
+            "row_status",
+            "timings_min_usecs",
+            "timings_max_usecs",
+            "timings_median_usecs",
+            "timings_mean_usecs",
+            "timings_stddev_usecs",
+            ])
+
+    def __init__(self):
+        self._line_items = []
+
+    def validate_user_supplied_kwargs(self, kwarg_dict):
+        name_conflicts = set(kwarg_dict.keys()).intersection(self.BUILTIN_COLUMN_NAMES)
+
+        if len(name_conflicts) > 0:
+            name_list = ', '.join(name_conflicts)
+            raise Exception(f'Attempting to supply values for buil-in column names: {name_list}')
+
+    def record_success(self, timings, **kwargs):
+        """
+        `timings` : Assumed to have the structure and syntax of
+          the timing results provided by TVM's `time_evaluator`
+          mechanism.
+
+        `kwargs` : Optional values for any of the other columns
+          defined for this benchmark table.
+        """
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+
+        line_item['row_status'] = 'SUCCESS'
+
+        line_item['timings_min_usecs'] = timings.min * 1000000
+        line_item['timings_max_usecs'] = timings.max * 1000000
+        line_item['timings_median_usecs'] = timings.median * 1000000
+        line_item['timings_stddev_usecs'] = timings.std * 1000000
+        line_item['timings_mean_usecs'] = timings.mean * 1000000
+
+        self._line_items.append(line_item)
+
+    def record_skip(self, **kwargs):
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+        line_item['row_status'] = 'SKIP'
+        self._line_items.append(line_item)
+
+    def record_fail(self, **kwargs):
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+        line_item['row_status'] = 'FAIL'
+        self._line_items.append(line_item)
+
+    def has_fail(self):
+        """
+        Returns True if the table contains at least one 'fail' line item,
+        otherwise returns False.
+        """
+        for li in self._line_items:
+            if li['row_status'] == 'FAIL':
+                return True
+
+        return False
+
+    def print_csv(self, f, column_name_order, timing_decimal_places=3):
+        """
+        Print the benchmark results as a csv.
+
+        `f` : The output stream.
+
+        `column_name_order`: an iterable sequence of column names, indicating the
+           order of column in the CSV output.
+
+           The CSV output will contain only those columns that are mentioned in
+           this list.
+
+        `timing_decimal_places`: for the numeric timing values, this is the
+           number of decimal places to provide in the printed output.
+           For example, a value of 3 is equivalent to the Python formatting string
+           `'{:.3f}'`
+        """
+
+        csv.register_dialect(
```

Comment:
Nit: Are there any differences between this dialect and the [`'excel-tab'`](https://docs.python.org/3/library/csv.html#csv.excel_tab) dialect?  If not, it would be easier to just use the existing dialect.

Reply:
Good catch - I think this was a vestige from when I was experimenting with different output formats.